### PR TITLE
Fixed issues #168 and #169

### PR DESCRIPTION
**Pull request by Yahasana**
if people don't want to render the main layout, safe to 
```
$this->view->setMainView(null);
$this->view->setContent('here u are');
```
or just 
```
$this->response->setContent('here u are');
```
in the end of the controller action

### CODE REVIEW

File: ice/mvc/app.zep [hunk 1]
```diff
@@ -93,7 +93,8 @@ class App extends Access
                     view = controller->view;
 
                 // Load views and set the response body if auto render
-                if this->autoRender {
+                // and there is no content in the response
+                if this->autoRender && response->getBody() === null {
```
Comment (mruz) [inline ice/mvc/app.zep:97]: Body is an empty string.
```zep
if this->autoRender && !response->getBody() {
```

File: ice/mvc/app.zep [hunk 2]
```diff
@@ -105,8 +106,11 @@ class App extends Access
 
                         view->setContent(view->render());
                     }
-
-                    response->setBody(view->layout(view->getMainView()));
+                    // if there is main layout view, other case e.g. load partial view through ajax
+                    let returned = view->getMainView();
+                    if returned {
+                        response->setBody(view->layout(returned));
```
Comment (mruz) [inline ice/mvc/app.zep]: We won't set the response body if no main view, maybe:
```zep
if view->getMainView() {
    response->setBody(view->layout(view->getMainView()));
} else {
    response->setBody(view->getContent());
}
```

**Review by mruz — CHANGES_REQUESTED** [review]

File: ice/mvc/app.zep (hunk 1 above)
Comment (Yahasana) [reply ice/mvc/app.zep:97]: if user want to send empty string `$this->response->setContent('');`

`response->getBody() === null` means user never touches `response->body` by hand and let it's decided by the system.

File: ice/mvc/app.zep (hunk 2 above)
Comment (Yahasana) [reply ice/mvc/app.zep]: You are right!

File: ice/mvc/app.zep (hunk 1 above)
Comment (mruz) [reply ice/mvc/app.zep:97]: Then we need `null` by default at https://github.com/ice/framework/blob/dev/ice/http/response.zep#L35
```zep
public function __construct(var body = null, int status = 200)
```

File: ice/http/response.zep [hunk 3]
```diff
@@ -32,7 +32,7 @@ class Response implements ResponseInterface
      * @param string body The HTTP response body
      * @param int status The HTTP response status
      */
-    public function __construct(string body = "", int status = 200)
+    public function __construct(string body = null, int status = 200)
```
Comment (mruz) [inline ice/http/response.zep]: One small thing change type `string` to `var`. 
`string body = null` it's casted to empty string by default

Comment (Yahasana) [reply ice/http/response.zep]: Good catch, fixed

**Review by mruz — APPROVED** [review]

**Pull request merged**